interfaces/builtin/tmpfs_mount.go

```diff
@@ -0,0 +1,77 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const tmpfsMountSummary = `allows mounting and unmounting TMPFS filesystems`
+
+const tmpfsMountBaseDeclarationSlots = `
+  tmpfs-mount:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const tmpfsMountConnectedPlugSecComp = `
+# Description: Allow mount and umount syscall access.
+mount
+umount
+umount2
+`
+
+const tmpfsMountConnectedPlugAppArmor = `
+# Description: Allow mounting and unmounting TMPFS filesystems.
+
+# Required for mounts and unmounts
+capability sys_admin,
+
+# Allow mounts to our snap-specific writable directories
+# parallel-installs: SNAP_{DATA,COMMON} are remapped, need to use SNAP_NAME, for
+# completeness allow SNAP_INSTANCE_NAME too
+mount fstype=tmpfs tmpfs -> /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/@{SNAP_REVISION}/{,**},
+mount fstype=tmpfs tmpfs -> /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/common/{,**},
+
+# NOTE: due to LP: #1613403, fstype is not mediated and as such, these rules
```

Comment:
It would be great if apparmor could mediate the filesystem type on unmount. I'm not sure if this is hard (missing infra in LSM) or easy (missing parser and state machine data). @alexmurray Is there any chance this bug could be re-visited, just to examine the complexity?

Reply:
@zyga this would be one for @jrjohansen to comment on I think.

Reply:
Indeed fstype= is not currently supported by the kernel for umount. It is not difficult to add, but it has not been a priority. This at earliest could be supported in 5.11 if it is decided its important enough to spend a day on.

Reply:
If there is a good reason it is possible we could deal with this as a bug and land it in 5.10 dev, and get it cherry-picked pick on the stable kernels.